terraform/modules/shared/main.tf

```diff
@@ -0,0 +1,376 @@
+  location            = var.location
+
+  subnet {
+    name           = "phdi-${terraform.workspace}-aks_subnet"
+    address_prefix = var.k8s_subnet_address_prefix
+  }
+}
+
+resource "azurerm_network_security_group" "aks_nsg" {
+  name                = "phdi-${terraform.workspace}-nsg"
+  location            = var.location
+  resource_group_name = var.resource_group_name
+
+  security_rule {
+    name                       = "AzureDataFactoryInbound"
+    priority                   = 100
+    direction                  = "Inbound"
+    access                     = "Allow"
+    protocol                   = "*"
+    source_port_range          = "*"
+    destination_port_range     = "*"
+    source_address_prefix      = "DataFactory"
+    destination_address_prefix = "*"
+  }
+  security_rule {
+    name                       = "AzureDataFactoryOutbound"
+    priority                   = 100
+    direction                  = "Outbound"
+    access                     = "Allow"
+    protocol                   = "*"
+    source_port_range          = "*"
+    destination_port_range     = "*"
+    source_address_prefix      = "*"
+    destination_address_prefix = "DataFactory"
+  }
+}
+
+resource "azurerm_subnet_network_security_group_association" "aks_nsg_association" {
+  subnet_id                 = azurerm_virtual_network.aks_vnet.subnet.*.id[0]
+  network_security_group_id = azurerm_network_security_group.aks_nsg.id
+}
+
+#### Kubernetes Service ####
+
+resource "azurerm_kubernetes_cluster" "cluster" {
+  name                = "phdi-${terraform.workspace}-cluster"
+  location            = var.location
+  resource_group_name = var.resource_group_name
+  dns_prefix          = "phdi-${terraform.workspace}"
+
+  default_node_pool {
+    name           = "default"
+    node_count     = 1
+    vm_size        = "Standard_D2_v2"
+    vnet_subnet_id = azurerm_virtual_network.aks_vnet.subnet.*.id[0]
+  }
+
+  identity {
+    type = "SystemAssigned"
+  }
+  network_profile {
+    network_plugin = "azure"
+  }
+}
+
+data "azurerm_kubernetes_cluster" "credentials" {
+  name                = azurerm_kubernetes_cluster.cluster.name
+  resource_group_name = var.resource_group_name
+}
+
+provider "helm" {
+  kubernetes {
+    host                   = data.azurerm_kubernetes_cluster.credentials.kube_config.0.host
+    client_certificate     = base64decode(data.azurerm_kubernetes_cluster.credentials.kube_config.0.client_certificate)
+    client_key             = base64decode(data.azurerm_kubernetes_cluster.credentials.kube_config.0.client_key)
+    cluster_ca_certificate = base64decode(data.azurerm_kubernetes_cluster.credentials.kube_config.0.cluster_ca_certificate)
+  }
+}
+
+resource "helm_release" "helm_clusters" {
+  for_each      = local.images
+  repository    = "https://cdcgov.github.io/phdi-charts/"
+  name          = "phdi-${terraform.workspace}-${each.key}"
+  chart         = "${each.key}-chart"
+  recreate_pods = true
+
```

Comment:
could you add this line here:

`set {
    name  = "storage_account_url"
    value = azurerm_storage_account.phi.primary_blob_endpoint
  }`